### ISSUE TITLE
Should we spec required prefixes directly in the relevant specs?

### ISSUE THREAD

**Issue by tabatkins** [w3c/csswg-drafts#247]
Some features have de-facto required prefixes.  This is currently specified in the Compat Spec, maintained in WHATWG.  Should we instead be specifying these directly in the spec?

My (possibly biased) reading of the issue:

Pro:
* keeps CSS features specced in CSS specs, rather than relying on an external standards body
* puts definitions of required features near the features they're related to, rather than in a completely separate spec - we purposely don't use errata specs for anything else!
* admits that these are, indeed, required parts of the web platform, even if they're distasteful

Cons:
* they're ugly I guess?

**Comment by upsuper** [comment]
Another con is that that would force implementations who did not need to face web-compatibility (e.g. epub) to implement the prefixed things as well to state their conformance, which sounds unfortunate.

**Comment by SebastianZ** [comment]
> - keeps CSS features specced in CSS specs, rather than relying on an external standards body
> - puts definitions of required features near the features they're related to, rather than in a completely separate spec - we purposely don't use errata specs for anything else!
> - admits that these are, indeed, required parts of the web platform, even if they're distasteful

Another option would be to move the specification to w3.org but keep it separated. Prefixed items may then be mentioned and linked to from the other specs.

A third con is that web authors should be encouraged to use the unprefixed items instead of the prefixed ones. (While the main audience of the specs are implementors, web authors also refer to them.)

Sebastian

**Comment by frivoal** [comment]
> A third con is that web authors should be encouraged to use the unprefixed items instead of the prefixed ones. (While the main audience of the specs are implementors, web authors also refer to them.)

This sort of depends how you write it. Yes, if the spec talks about it at all, that makes it more likely to be known than if it doesn't, but if you keep it low keep like [CSS-UI does with `user-select`](https://drafts.csswg.org/css-ui-4/#content-selection), it's hardly an encouragement.

**Comment by tabatkins** [comment]
Yeah, the prefixes should *clearly* be treated like any other deprecated-but-required feature, and be defined either in a compatibility appendix (like Colors does with the system colors, or MQ with some of the deprecated MQs) or just with a low-key paragraph near the end of the section (like UI, or Images' treatment of the SVG image-rendering values).

**Comment by tabatkins** [comment]
Group consensus on the call today is that we don't want to actually list the required prefixes in each spec, as the set does shift and change over time as compat data changes. We're also okay with WHATWG maintaining the spec for now, as long as they're willing to do so.

However, we do want to make it more obvious, so we'll:

1. Link to the compat spec from the snapshot.
2. Link to the compat spec in each spec that is featured in it.

**Comment by frivoal** [comment]
I don't really like that resolution, but I guess I can live with it. However, I have a few questions about what it means practically:

1) Does that mean we should remove the prefixes from our specs? -webkit-user-select was removed from the compat spec once it was added to ours, as that is (as far as I understand) the way the compat spec deals with things that have graduated into their main spec

2) What about other non standard names that don't have a prefix? The standard `overflow-wrap` property has a `word-wrap` initially proprietary name, which is now supported by multiple browsers. Except for the fact that the name doesn't include the word webkit, the situation is pretty much the same. Should we keep it in our spec, or move it to the compat spec?

3) The compat spec does not specify how aliasing works, it just says you need to alias. Our specs say: “must treat as a shorthand” for `word-wrap`, and “up to the UA, but should consider doing it as a shorthand” for `-webkit-user-select`. Do we preserve that information? Where?
